Integration of perftest CI with standardised benchmarks

vrnimje opened the pull request:
## Proposed Changes
  - Improved the perftest CI with CDash integration
  - Added separate cmake function for adding performance tests with specific options for benchmarking purposes, which also creates a custom target for generating the plots

## Any background context you want to provide?
Extension to #6448 
Sample run on CDash: https://my.cdash.org/builds/2589647

vrnimje committed:
Added initial perftests CI config for displaying plots in CDash

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added initial perftests CI config for displaying plots in CDash

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Configured ctest.cmake into CI, instead of directly using ctest -D

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Configured ctest.cmake into CI, instead of directly using ctest -D

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Moved plot generation to ctest.cmake

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Moved plot generation to ctest.cmake

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Changed some internal paths for json file generation

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Changed some internal paths for json file generation

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Fixed issue, where python script was not executing from ctest.cmake

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Fixed issue, where python script was not executing from ctest.cmake

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Adding custom target for cdash results and plot generation, for each perftest

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Adding custom target for cdash results and plot generation, for each perftest

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Separate cmake function defined for perftest reports, for taking additonal params

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Separate cmake function defined for perftest reports, for taking additonal params

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added more perf-report tests, and made improvements to the perftest handling

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added more perf-report tests, and made improvements to the perftest handling

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removing debgugging modifications

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removing debgugging modifications

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removed the cdash image printing for detailed config of perftest report

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removed the cdash image printing for detailed config of perftest report

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Defined the cmake config command for cdash properly

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Defined the cmake config command for cdash properly

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Fixing syntax for perftest_init in standardised benchmarks

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Fixing syntax for perftest_init in standardised benchmarks

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removing unnecessary perftest_cfg call from future_overhead_report test

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removing unnecessary perftest_cfg call from future_overhead_report test

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Update the plotting script

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Update the plotting script

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added comparision with pre-defined baselines, and standardised transform_reduce_scaling and benchmark_is_heap_until

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added comparision with pre-defined baselines, and standardised transform_reduce_scaling and benchmark_is_heap_until

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Standardised benchmark_inplace_merge and benchmark_merge

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Standardised benchmark_inplace_merge and benchmark_merge

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Standardised benchmark_is_heap and benchmark_remove

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Standardised benchmark_is_heap and benchmark_remove

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Standardised benchmark_remove_if and fixed formatting issues with benchmark results

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Standardised benchmark_remove_if and fixed formatting issues with benchmark results

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje commented:
<details><summary>Performance test report</summary><h3>small_vector_benchmark</h3><ol><li><b>hpx::small_vector</b>, &lt;fxn&lt;void()&gt;, 2&gt;: Performance is better by 0.46 % (KS-stat: 0.9583333333333334)</li></li></ol><h3>stream_report</h3><ol><li><b>Stream benchmark - Scale</b>, fork_join_executor_default_fork_join_policy_allocator: Performance is better by 12.69 % (KS-stat: 0.9583333333333334)</li></li><li><b>Stream benchmark - Triad</b>, fork_join_executor_default_fork_join_policy_allocator: Performance is better by 3.25 % (KS-stat: 0.875)</li></li><li><b>Stream benchmark - Scale</b>, scheduler_executor_default_scheduler_executor_allocator: Performance is better by 18.40 % (KS-stat: 1.0)</li></li><li><b>Stream benchmark - Add</b>, scheduler_executor_default_scheduler_executor_allocator: Performance is better by 10.84 % (KS-stat: 0.7916666666666666)</li></li></ol><h3>foreach_report</h3><ol><li><b>for_each</b>, fork_join_executor: Performance is better by 6.67 % (KS-stat: 0.9166666666666666)</li></li></ol><h3>transform_reduce_scaling</h3><ol><li><b>transform_reduce</b>, par: Performance is better by 57.55 % (KS-stat: 1.0)</li></li><li><b>transform_reduce_old</b>, par: Performance is better by 69.66 % (KS-stat: 1.0)</li></li></ol><h3>benchmark_is_heap_until</h3><ol><li><b>hpx::heap_until</b>, seq: Performance is better by 35.65 % (KS-stat: 1.0)</li></li><li><b>hpx::heap_until</b>, par: Performance is better by 32.25 % (KS-stat: 0.9583333333333334)</li></li></ol><h3>benchmark_inplace_merge</h3><ol><li><b>hpx::inplace_merge</b>, seq: Performance is better by 41.21 % (KS-stat: 0.9166666666666666)</li></li><li><b>hpx::inplace_merge</b>, par_unseq: Performance is better by 49.60 % (KS-stat: 1.0)</li></li></ol><h3>benchmark_is_heap</h3><ol><li><b>hpx::is_heap</b>, seq: Performance is better by 100.00 % (KS-stat: 1.0)</li></li></ol><h3>benchmark_remove</h3><ol><li><b>hpx::remove</b>, seq: Performance is better by 21.37 % (KS-stat: 1.0)</li></li><li><b>hpx::remove</b>, par: Performance is better by 30.38 % (KS-stat: 0.9583333333333334)</li></li><li><b>hpx::remove</b>, par_unseq: Performance is better by 31.96 % (KS-stat: 0.9166666666666666)</li></li></ol><h3>benchmark_remove_if</h3><ol><li><b>hpx::remove_if</b>, seq: Performance is better by 32.02 % (KS-stat: 1.0)</li></li><li><b>hpx::remove_if</b>, par: Performance is better by 38.20 % (KS-stat: 1.0)</li></li><li><b>hpx::remove_if</b>, par_unseq: Performance is better by 37.82 % (KS-stat: 1.0)</li></li></ol><h3>minmax_element_performance</h3><ol><li><b>hpx::min</b>, par: Performance is better by 19.53 % (KS-stat: 0.9166666666666666)</li></li></ol><h3>small_vector_benchmark</h3><ol><li><b>hpx::small_vector</b>, &lt;fxn&lt;void()&gt;, 1&gt;: Performance is worse by 9.37 % (KS-stat: 0.7916666666666666)</li></li><li><b>hpx::small_vector</b>, &lt;fxn&lt;void()&gt;, 8&gt;: Performance is better by 1.68 % (KS-stat: 0.9166666666666666)</li></li></ol><h3>stream_report</h3><ol><li><b>stream benchmark - Copy</b>, fork_join_executor_default_fork_join_policy_allocator: Performance is better by 6.28 % (KS-stat: 1.0)</li></li><li><b>Stream benchmark - Scale</b>, fork_join_executor_default_fork_join_policy_allocator: Performance is better by 13.83 % (KS-stat: 1.0)</li></li><li><b>Stream benchmark - Add</b>, fork_join_executor_default_fork_join_policy_allocator: Performance is better by 4.41 % (KS-stat: 0.7916666666666666)</li></li><li><b>Stream benchmark - Triad</b>, fork_join_executor_default_fork_join_policy_allocator: Performance is better by 2.72 % (KS-stat: 0.75)</li></li><li><b>Stream benchmark - Scale</b>, scheduler_executor_default_scheduler_executor_allocator: Performance is better by 18.22 % (KS-stat: 0.9583333333333334)</li></li></ol><h3>foreach_report</h3><ol><li><b>for_each</b>, fork_join_executor: Performance is better by 6.40 % (KS-stat: 0.9166666666666666)</li></li></ol><h3>transform_reduce_scaling</h3><ol><li><b>transform_reduce</b>, par: Performance is better by 57.40 % (KS-stat: 1.0)</li></li><li><b>transform_reduce_old</b>, par: Performance is better by 69.81 % (KS-stat: 1.0)</li></li></ol><h3>benchmark_is_heap_until</h3><ol><li><b>hpx::heap_until</b>, seq: Performance is better by 36.00 % (KS-stat: 1.0)</li></li><li><b>hpx::heap_until</b>, par: Performance is better by 32.02 % (KS-stat: 0.9583333333333334)</li></li></ol><h3>benchmark_inplace_merge</h3><ol><li><b>hpx::inplace_merge</b>, seq: Performance is better by 41.25 % (KS-stat: 0.9583333333333334)</li></li><li><b>hpx::inplace_merge</b>, par_unseq: Performance is better by 49.76 % (KS-stat: 1.0)</li></li></ol><h3>benchmark_is_heap</h3><ol><li><b>hpx::is_heap</b>, seq: Performance is better by 100.00 % (KS-stat: 1.0)</li></li></ol><h3>benchmark_remove</h3><ol><li><b>hpx::remove</b>, seq: Performance is better by 21.60 % (KS-stat: 0.9583333333333334)</li></li><li><b>hpx::remove</b>, par: Performance is better by 31.05 % (KS-stat: 1.0)</li></li><li><b>hpx::remove</b>, par_unseq: Performance is better by 33.81 % (KS-stat: 1.0)</li></li></ol><h3>benchmark_remove_if</h3><ol><li><b>hpx::remove_if</b>, seq: Performance is better by 29.03 % (KS-stat: 1.0)</li></li><li><b>hpx::remove_if</b>, par: Performance is better by 37.92 % (KS-stat: 1.0)</li></li><li><b>hpx::remove_if</b>, par_unseq: Performance is better by 37.52 % (KS-stat: 1.0)</li></li></ol><h3>small_vector_benchmark</h3><ol><li><b>hpx::small_vector</b>, &lt;int, 1&gt;: Performance is worse by 2.90 % (KS-stat: 0.7916666666666666)</li></li><li><b>hpx::small_vector</b>, &lt;fxn&lt;void()&gt;, 1&gt;: Performance is better by 8.86 % (KS-stat: 0.7916666666666666)</li></li></ol><h3>stream_report</h3><ol><li><b>stream benchmark - Copy</b>, fork_join_executor_default_fork_join_policy_allocator: Performance is better by 1.72 % (KS-stat: 0.7916666666666666)</li></li></ol><h3>minmax_element_performance</h3><ol><li><b>hpx::min</b>, par: Performance is worse by 89.74 % (KS-stat: 0.75)</li></li></ol><h3>stream_report</h3><ol><li><b>stream benchmark - Copy</b>, fork_join_executor_default_fork_join_policy_allocator: Performance is worse by 3.32 % (KS-stat: 1.0)</li></li><li><b>Stream benchmark - Scale</b>, fork_join_executor_default_fork_join_policy_allocator: Performance is worse by 21.33 % (KS-stat: 0.8333333333333334)</li></li><li><b>Stream benchmark - Add</b>, fork_join_executor_default_fork_join_policy_allocator: Performance is worse by 16.76 % (KS-stat: 0.7916666666666666)</li></li><li><b>Stream benchmark - Triad</b>, fork_join_executor_default_fork_join_policy_allocator: Performance is worse by 3.95 % (KS-stat: 0.75)</li></li><li><b>Stream benchmark - Add</b>, scheduler_executor_default_scheduler_executor_allocator: Performance is better by 15.89 % (KS-stat: 0.9166666666666666)</li></li></ol><h3>benchmark_remove_if</h3><ol><li><b>hpx::remove_if</b>, seq: Performance is better by 4.40 % (KS-stat: 0.75)</li></li></ol></details>

Pansysk75 commented:
@vrnimje Good job! ;)

vrnimje committed:
Added KS-test for benchmark comparision, and updated baselines

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added KS-test for benchmark comparision, and updated baselines

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Made modifications for perftests comment mechanism

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Made modifications for perftests comment mechanism

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removed the reports folder from the artifacts, and clang-formatted files

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removed the reports folder from the artifacts, and clang-formatted files

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Moved comment mechanism to entry.sh

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Moved comment mechanism to entry.sh

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Sourcing the environment file, and removed boost fetch from perftests' ctest

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Sourcing the environment file, and removed boost fetch from perftests' ctest

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added asio fetch option for CMake Configure step in perftests' ctest.cmake

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added asio fetch option for CMake Configure step in perftests' ctest.cmake

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Manually installing scipy module for plotting script

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Manually installing scipy module for plotting script

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Using user installation for scipy, due to permission issues

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Using user installation for scipy, due to permission issues

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removed scipy installation, as it has been added to virtual env

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removed scipy installation, as it has been added to virtual env

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Importing scipy.stats in plotting script

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Importing scipy.stats in plotting script

Signed-off-by: Vedant <vedantnimjed@gmail.com>

Pansysk75 commented:
retest lsu perf

Pansysk75 commented:
retest lsu perf

vrnimje committed:
Removed pushd commands in comment_github.sh

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removed pushd commands in comment_github.sh

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Re-formatted comment for performance discrepancy, and reverted back to HPX CDash creds

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Re-formatted comment for performance discrepancy, and reverted back to HPX CDash creds

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removed addition of newlines into perf-report, and referenced correct variables

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removed addition of newlines into perf-report, and referenced correct variables

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added commands to write CTest Build ID to the file, in ctest.cmake

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added commands to write CTest Build ID to the file, in ctest.cmake

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Calling status_computation function manually in lsu-perftest, and mentioned unused variables in benchmark_is_heap_until

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Calling status_computation function manually in lsu-perftest, and mentioned unused variables in benchmark_is_heap_until

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added option for printing CDash image path, and modified lsu-perftest for uploading results JSONs

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added option for printing CDash image path, and modified lsu-perftest for uploading results JSONs

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Updated perftest-baselines, moved comment mechanism to status function and fixed formatting

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Updated perftest-baselines, moved comment mechanism to status function and fixed formatting

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removed explicit call to artifacts-storage function, and modified perftest-comment format

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removed explicit call to artifacts-storage function, and modified perftest-comment format

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removing ctest_status check for commenting perftest-report

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removing ctest_status check for commenting perftest-report

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Changed plotting format, and made definitions of dependencies and executables for cdash targets implicit

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Changed plotting format, and made definitions of dependencies and executables for cdash targets implicit

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Standardised the following tests, and made formatting changes to plotting script
* benchmark_partial_sort
* benchmark_partial_sort_parallel
* benchmark_nth_element
* benchmark_nth_element_parallel

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Standardised the following tests, and made formatting changes to plotting script
* benchmark_partial_sort
* benchmark_partial_sort_parallel
* benchmark_nth_element
* benchmark_nth_element_parallel

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Fixed subplot indexing issue in perftest plotting script, updated baselines, and fixed clang- and cmake-formatting issues

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Fixed subplot indexing issue in perftest plotting script, updated baselines, and fixed clang- and cmake-formatting issues

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Scaling down perftest plot images, so that it is displayed entirely on CDash, and added them to the build artifacts

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Scaling down perftest plot images, so that it is displayed entirely on CDash, and added them to the build artifacts

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Changed plotting format orientation

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Changed plotting format orientation

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Increasing test output size for CDash, using CTestCustom.ctest file

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Increasing test output size for CDash, using CTestCustom.ctest file

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Renamed ctestcustom file with cmake extension, and resized image plots again

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Renamed ctestcustom file with cmake extension, and resized image plots again

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Increased number of epochs for performance measurements

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Increased number of epochs for performance measurements

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Updated baselines with updated number of epochs, and changed uploaded image name for CDash

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Updated baselines with updated number of epochs, and changed uploaded image name for CDash

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Separated comparision subplot images, and changed bootstrap statistic to use medians

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Separated comparision subplot images, and changed bootstrap statistic to use medians

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added measurements for each sub-test on CDash, and prefixed performance cmdline options with hpx:

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added measurements for each sub-test on CDash, and prefixed performance cmdline options with hpx:

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removed escape characters from the CDash mustache template

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Removed escape characters from the CDash mustache template

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added options for test_count in the tests missing it and increased internal benchmarking iterations to 1000

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Added options for test_count in the tests missing it and increased internal benchmarking iterations to 1000

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Updated perftest baselines to have 1000 iterations, and cleaned up perftest_plot script

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Updated perftest baselines to have 1000 iterations, and cleaned up perftest_plot script

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Modified bootstrapping parameters, and reduced the perftest iterations to 100

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Modified bootstrapping parameters, and reduced the perftest iterations to 100

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Changed plots to histogram with kde, removed generated images from artifacts, and updated baselines to a stable set

Signed-off-by: Vedant <vedantnimjed@gmail.com>

vrnimje committed:
Changed plots to histogram with kde, removed generated images from artifacts, and updated baselines to a stable set

Signed-off-by: Vedant <vedantnimjed@gmail.com>

hkaiser commented:
@vrnimje Sorry it took us so long to get back to your PR. I'll try to finalize it over the next few days or so.

hkaiser committed:
Updating reference data for rostam

- flyby: increase number of iterations for performance tests to reduce variance

Signed-off-by: Hartmut Kaiser <hartmut.kaiser@gmail.com>

hkaiser committed:
Updating reference data for rostam

- flyby: increase number of iterations for performance tests to reduce variance

Signed-off-by: Hartmut Kaiser <hartmut.kaiser@gmail.com>

hkaiser committed:
Merge branch 'master' into cdash_perftest_bench

hkaiser committed:
Addressing review comments

Signed-off-by: Hartmut Kaiser <hartmut.kaiser@gmail.com>

hkaiser committed:
Merge branch 'cdash_perftest_bench' of github.com:vrnimje/hpx-fork into cdash_perftest_bench

Signed-off-by: Hartmut Kaiser <hartmut.kaiser@gmail.com>

hkaiser committed:
More review comments addressed

Signed-off-by: Hartmut Kaiser <hartmut.kaiser@gmail.com>

hkaiser committed:
Merge branch 'master' into cdash_perftest_bench

Signed-off-by: Hartmut Kaiser <hartmut.kaiser@gmail.com>
